src/raster/rasterband.rs

```diff
@@ -950,13 +950,8 @@ impl<'a> RasterBand<'a> {
     /// Return the unit of the rasterband.
     /// If there is no unit, the empty string is returned.
     pub fn unit(&self) -> String {
-        let str_ptr = unsafe { gdal_sys::GDALGetRasterUnitType(self.c_rasterband) };
-
-        if str_ptr.is_null() {
-            return String::new();
-        }
-
-        _string(str_ptr)
+        let c_ptr = unsafe { gdal_sys::GDALGetRasterUnitType(self.c_rasterband) };
```

Comment:
`c_ptr` is owned by `self` so we don't need to free it?

Reply:
Owned by `GDALRasterBand`.

> The returned string should not be modified, nor freed by the calling application.